Util/InformationTheory/src/main/java/org/tribuo/util/infotheory/InformationTheory.java

```diff
@@ -538,6 +552,53 @@ public static double calculateEntropy(DoubleStream vector) {
         return vector.map((p) -> (- p * Math.log(p) / LOG_BASE)).sum();
     }
 
+    /**
+     * Compute the expected mutual information assuming randomized inputs.
+     *
+     * @param first The first vector.
+     * @param second The second vector.
+     * @return The expected mutual information under a hypergeometric distribution.
+     */
+    public static <T> double expectedMI(List<T> first, List<T> second) {
```

Comment:
This method was formerly a private method in ClusteringMetrics and is now a public method here - seems like it deserves a unit test.  Three options seems reasonable - generate some test data using a method from a similar library (suggestions welcome!), generate some regression data with this method and test on that (tautological, I suppose, but at least you will know if the behavior changes), or work out some simple examples by hand.  Thoughts?

Reply:
Sure, I can work out some examples to test.